src/dftbp/dftb/periodic.F90

```diff
@@ -460,16 +481,37 @@ subroutine updateNeighbourList(coord, img2CentCell, iCellVec, neigh, nAllAtom, c
     cutoff2 = cutoff**2
     nAllAtom = 0
 
+    parallel = .false.
+#:if WITH_MPI
+    parallel = present(env)
+#:endif
+
+    if (parallel) then
+#:if WITH_MPI
+      call distributeAtoms(env%mpi%nodeComm%rank, env%mpi%nodeComm%size, nAtom, &
+          & startAtom, endAtom, parallelSetupError)
+#:else
+      parallelSetupError = .true.
+#:endif
+      if (parallelSetupError) then
+        parallel = .false.
+      end if
+    end if
+
+    if (.not. parallel) then
```

Comment:
Is there a way to unify with  the existing MPI [distribution routine](https://github.com/dftbplus/dftbplus/blob/199d8cf7ea1aba266b10b0d12607bd37e592cc34/src/dftbp/common/schedule.F90#L39)  for similar loops?

Reply:
Thanks for pointing this out. I will have a look into this. Could well be that those routines already do what I need here.